include/net/net_pkt.h

```diff
@@ -95,8 +95,11 @@ struct net_pkt {
 	u8_t family     : 4;	/* IPv4 vs IPv6 */
 	u8_t _unused    : 3;
 
+	u8_t ip_hop_limit;	/* IPv6 hop limit or IPv4 ttl for this network
+				 * packet. The value is shared between IPv6/4
```

Comment:
Well, let's have a union here:
union {
        u8_t ipv6_hop_limit;
        u8_t ipv4_ttl;
};

It's self documented then, and also make more sens in functions below.

Reply:
Sure, good idea.